zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/EventFromIncidentHandler.java

```diff
@@ -0,0 +1,222 @@
+            record.getValue().getElementInstanceKey()));
+  }
+
+  @Override
+  public EventEntity createNewEntity(final String id) {
+    return new EventEntity().setId(id);
+  }
+
+  @Override
+  public void updateEntity(final Record<IncidentRecordValue> record, final EventEntity entity) {
+
+    final IncidentRecordValue recordValue = record.getValue();
+    entity
+        .setId(
+            String.format(
+                ID_PATTERN,
+                recordValue.getProcessInstanceKey(),
+                recordValue.getElementInstanceKey()))
+        .setPositionIncident(record.getPosition());
+    loadEventGeneralData(record, entity);
+
+    if (recordValue.getProcessInstanceKey() > 0) {
+      entity.setProcessInstanceKey(recordValue.getProcessInstanceKey());
+    }
+    entity
+        .setBpmnProcessId(recordValue.getBpmnProcessId())
+        .setFlowNodeId(recordValue.getElementId())
+        .setTenantId(tenantOrDefault(recordValue.getTenantId()));
+    if (recordValue.getElementInstanceKey() > 0) {
+      entity.setFlowNodeInstanceKey(recordValue.getElementInstanceKey());
+    }
+
+    final EventMetadataEntity eventMetadata = new EventMetadataEntity();
+    eventMetadata.setIncidentErrorMessage(trimWhitespace(recordValue.getErrorMessage()));
+    eventMetadata.setIncidentErrorType(
+        ErrorType.fromZeebeErrorType(
+            recordValue.getErrorType() == null ? null : recordValue.getErrorType().name()));
+
+    entity.setMetadata(eventMetadata);
+  }
+
+  @Override
+  public void flush(final EventEntity entity, final BatchRequest batchRequest) {
+    persistEvent(entity, POSITION_INCIDENT, entity.getPositionIncident(), batchRequest);
+  }
+
+  private void loadEventGeneralData(final Record record, final EventEntity eventEntity) {
+    eventEntity.setKey(record.getKey());
+    eventEntity.setPartitionId(record.getPartitionId());
+    eventEntity.setEventSourceType(
+        EventSourceType.fromZeebeValueType(
+            record.getValueType() == null ? null : record.getValueType().name()));
+    eventEntity.setDateTime(
+        OffsetDateTime.ofInstant(Instant.ofEpochMilli(record.getTimestamp()), UTC));
+    eventEntity.setEventType(EventType.fromZeebeIntent(record.getIntent().name()));
+  }
+
+  private void persistEvent(
+      final EventEntity entity,
+      final String positionFieldName,
+      final long positionFieldValue,
+      final BatchRequest batchRequest) {
+
+    final Map<String, Object> jsonMap = new HashMap<>();
+    jsonMap.put(KEY, entity.getKey());
+    jsonMap.put(EVENT_SOURCE_TYPE, entity.getEventSourceType());
+    jsonMap.put(EVENT_TYPE, entity.getEventType());
+    jsonMap.put(DATE_TIME, entity.getDateTime());
+    jsonMap.put(PROCESS_KEY, entity.getProcessDefinitionKey());
+    jsonMap.put(BPMN_PROCESS_ID, entity.getBpmnProcessId());
+    jsonMap.put(FLOW_NODE_ID, entity.getFlowNodeId());
+    jsonMap.put(positionFieldName, positionFieldValue);
+    if (entity.getMetadata() != null) {
```

Comment:
I don't see unit tests for updating the metadata fields. Could we may be add couple of fields for every record type?

Reply:
it is there, the last few lines in `testUpdateEntity` in each handler test

Reply:
or do you mean on flush?

Reply:
done